ruff.toml

```diff
@@ -36,9 +35,14 @@ ignore = [
     "E721",
     "E731",
     "E741",
+    "F403",  # `from ... import *` used; unable to detect undefined names
+    "F405",  # may be undefined, or defined from star imports
+    "F821",  # Undefined name
+    "F841",  # Local variable is assigned to but never used
     "UP015", # Unnecessary mode argument
     "UP031", # TODO: Use format specifiers instead of percent format
 ]
 
 [lint.per-file-ignores]
 "test*.py" = ["E201", "E714"]
+"__init__.py" = ["F401"]
```

Comment:
usually you inline `# ruff: noqa` this one, so why the exception?

Reply:
I don't know. Ignoring F401 in `__init__.py` files is pretty common, so I added this upfront. The rest is the result of gradually changing "too many" lines with `noqa:` into a single `ruff: noqa:`. I agree I should be consistent. Which style do you prefer? In the file itself or `pyproject.toml`?